macros/model_functions/base_ga4_event_postgres_query.sql

```
@@ -0,0 +1,311 @@
+      , {{ property }}->'web_info'->>'hostname' AS {{ property }}_web_info_hostname
+
+    {%- elif property == 'ecommerce' %}
+      , {{ property }}->>'total_item_quantity' AS {{ property }}_total_item_quantity
+      , {{ property }}->>'purchase_revenue_in_usd' AS {{ property }}_purchase_revenue_in_usd
+      , {{ property }}->>'purchase_revenue' AS {{ property }}_purchase_revenue
+      , {{ property }}->>'refund_value_in_usd' AS {{ property }}_refund_value_in_usd
+      , {{ property }}->>'refund_revenue' AS {{ property }}_refund_value
+      , {{ property }}->>'shipping_value_in_usd' AS {{ property }}_shipping_value_in_usd
+      , {{ property }}->>'shipping_value' AS {{ property }}_shipping_value
+      , {{ property }}->>'tax_value_in_usd' AS {{ property }}_tax_value_in_usd
+      , {{ property }}->>'tax_value' AS {{ property }}_tax_value
+      , {{ property }}->>'unique_items' AS {{ property }}_unique_items
+      , {{ property }}->>'transaction_id' AS {{ property }}_transaction_id
+
+    {%- elif property == 'event_dimensions' %}
+      , {{ property }}
+
+    {%- elif property == 'geo' %}
+      , {{ property }}->>'city' AS {{ property }}_city
+      , {{ property }}->>'metro' AS {{ property }}_metro
+      , {{ property }}->>'region' AS {{ property }}_region
+      , {{ property }}->>'country' AS {{ property }}_country
+      , {{ property }}->>'continent' AS {{ property }}_continent
+      , {{ property }}->>'sub_continent' AS {{ property }}_sub_continent
+
+    {%- elif property == 'traffic_source' %}
+      , {{ property }}->>'name' AS {{ property }}_campaign
+      , {{ property }}->>'medium' AS {{ property }}_medium
+      , {{ property }}->>'source' AS {{ property }}_source
+
+    {%- elif property == 'user_ltv' %}
+      , {{ property }}->>'revenue' AS {{ property }}_revenue
+      , {{ property }}->>'currency' AS {{ property }}_currency
+
+    {%- endif -%}
```

Question (blocking): What if there are other properties?

Is that not allowed? Then let's throw a descriptive error. Otherwise, Let's capture it somehow.